app/src/main/java/com/onboarding/parkingsystemjava/mvp/presenter/ReservePresenter.java

```diff
@@ -22,19 +28,44 @@ public void onEndDateButtonPress() {
 
     @Override
     public void onOkButtonPress() {
-        view.returnToMainActivity();
+        model.setReservationInfo(view.getParkingLot(), view.getUserPassword());
+        Reservation reservation = model.getReservation();
+        ReserveComprobation comprobation = model.reserveFieldsCheck(reservation);
+        switch (comprobation) {
+            case MISSING_START:
+                view.showMissingStartDateToast();
+                break;
+            case MISSING_END:
+                view.showMissingEndDateToast();
+                break;
+            case MISSING_LOT:
+                view.showMissingParkingLotToast();
+                break;
+            case MISSING_PASSWORD:
+                view.showMissingUserPasswordToast();
+                break;
+            case COMPROBATION_OK:
+                view.showReserveSavedToast();
+                break;
+        }
+        if (comprobation == ReserveComprobation.COMPROBATION_OK) {
+            model.addReservation(reservation);
+            view.returnToMainActivity();
+        }
     }
 
     @Override
     public void onCancelButtonPress() {
         view.returnToMainActivity();
     }
 
-    public void setStartDateTextView(String date) {
+    public void setReservationStartDate(Calendar startDate, String date) {
+        model.setStartDate(startDate);
         view.setStartDateTextView(date);
```

Comment:
here you should use the info from the model
`        view.setStartDateTextView(model.getStartDate());`

Reply:
done